apps/vscode/src/main.ts

```diff
@@ -117,6 +124,9 @@ export function activate(c: ExtensionContext) {
       manuallySelectWorkspaceDefinitionCommand
     );
 
+    // registers itself as a CodeLensProvider and watches config to dispose/re-register
```

Comment:
How's this? I have this registering itself as a CodeLensProvider and watching the config settings for changes to dispose/re-register?